flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceReader.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source.reader;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.flink.api.connector.source.SourceReaderContext;
+import org.apache.flink.connector.base.source.reader.SingleThreadMultiplexSourceReaderBase;
+import org.apache.iceberg.flink.source.IcebergSourceEvents;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplit;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+public class IcebergSourceReader<T> extends
+    SingleThreadMultiplexSourceReaderBase<RecordAndPosition<T>, T, IcebergSourceSplit, IcebergSourceSplit> {
+
+  public IcebergSourceReader(
+      ReaderFunction<T> readerFunction,
+      SourceReaderContext context,
+      IcebergSourceReaderMetrics metrics) {
+    super(
+        () -> new IcebergSourceSplitReader<>(readerFunction, context, metrics),
+        new IcebergSourceRecordEmitter<>(),
+        context.getConfiguration(),
+        context);
+  }
+
+  @Override
+  public void start() {
+    requestSplit(Collections.emptyList());
+  }
+
+  @Override
+  protected void onSplitFinished(Map<String, IcebergSourceSplit> finishedSplitIds) {
+    if (!finishedSplitIds.isEmpty()) {
```

Comment:
When would this be empty? And should this still call `requestSplit` even if there is no finished split ID?

Reply:
you are right. checked the base class code. it won't be empty. will remove the unnecessary if check